cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -383,16 +388,51 @@ func ValidateIPNetFromString(subnet string, minAddrs int64, fldPath *field.Path)
 	return allErrs
 }
 
-// ValidateNetworking validates networking configuration
+// ValidateIPNetFromStringForDualStack validates network portion of IP addresses for dual-stack configurations
+func ValidateIPNetFromStringForDualStack(subnet string, minAddrs int64, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	subnets := strings.Split(subnet, ",")
+	if len(subnets) < 2 {
+		allErrs = append(allErrs, field.Invalid(fldPath, subnet, "expected more than 1 pod subnet in dual-stack configuration"))
+	}
+	for _, subNetStr := range subnets {
+		_, subNet, err := net.ParseCIDR(strings.TrimSpace(subNetStr))
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, subNetStr, "couldn't parse subnet"))
+		} else {
+			numAddresses := ipallocator.RangeSize(subNet)
+			if numAddresses < minAddrs {
+				allErrs = append(allErrs, field.Invalid(fldPath, subnet, "subnet is too small"))
+			}
+		}
+	}
+	return allErrs
+}
+
+// isDualStackNetworking returns true if more than one pod CIDR has been specified
+func isDualStackNetworking(c *kubeadm.Networking) bool {
+	cidrs := strings.Split(c.PodSubnet, ",")
+	return len(cidrs) > 1
+}
+
+// ValidateNetworking validates networking configuration for the (original) single-stack case.
 func ValidateNetworking(c *kubeadm.Networking, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	dnsDomainFldPath := field.NewPath("dnsDomain")
 	for _, err := range validation.IsDNS1123Subdomain(c.DNSDomain) {
 		allErrs = append(allErrs, field.Invalid(dnsDomainFldPath, c.DNSDomain, err))
 	}
-	allErrs = append(allErrs, ValidateIPNetFromString(c.ServiceSubnet, constants.MinimumAddressesInServiceSubnet, field.NewPath("serviceSubnet"))...)
-	if len(c.PodSubnet) != 0 {
-		allErrs = append(allErrs, ValidateIPNetFromString(c.PodSubnet, constants.MinimumAddressesInServiceSubnet, field.NewPath("podSubnet"))...)
+	if isDualStackNetworkingConfig {
```

Comment:
on the same line of [2] from @neolit123 
I have some concern about this approach...
What if someone calls `ValidateNetworking` directly? it will always assume `isDualStackNetworkingConfig == false`

Reply:
Done!